File: src/main/java/io/github/darleywey/honoendpoints/model/HonoEndpoint.java
package io.github.darleywey.honoendpoints.model;

import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;

public record HonoEndpoint(
        @NotNull String method,
        @NotNull String path,
        @NotNull PsiElement source
) {}

File: src/main/kotlin/io/github/darleywey/honoendpoints/model/HonoEndpoint.kt
package io.github.darleywey.honoendpoints.model

import com.intellij.psi.PsiElement

/** [source] retains the full call; [target] anchors navigation to its path literal. */
data class HonoEndpoint(
    val method: String,
    val path: String,
    val source: PsiElement,
    val target: PsiElement,
)
